tools/metadata/webfeatures/schema.py

```diff
@@ -0,0 +1,86 @@
+from enum import Enum
+from dataclasses import dataclass
+from fnmatch import fnmatchcase
+from typing import Any, Dict, Sequence, Union
+
+from ..schema import SchemaValue, validate_dict
+
+"""
+YAML filename for meta files
+"""
+WEB_FEATURES_YML_FILENAME = "WEB_FEATURES.yml"
+
+
+class SpecialFileEnum(Enum):
+    """All files recursively"""
+    RECURSIVE = "**"
+
+
+class FeatureFile(str):
+    def match_files(self, base_filenames: Sequence[str]) -> Sequence[str]:
+        """
+        Given the input base file names, returns the subset of base file names
```

Comment:
I'd like to call out a complication that will come up. IIUC the lint rule is that the file name must match. The file name might be `blob-range.any.js`. On wpt.fyi however we're dealing with test names, which might be `blob-range.any.html` and `blob-range.any.worker.html`. One file might produce multiple tests.

Have you decided where the mapping from file names to test names will happen? Doing that requires using the manifest of the correct version of WPT. I think the candidate places to do it would be while building the `WEB_FEATURES_MANIFEST.json` file, or in wpt.fyi when processing `WEB_FEATURES_MANIFEST.json` into the data needed for the `feature:` search atom.

Reply:
The initial approach I had in mind was to do it while building WEB_FEATURES_MANIFEST.json.

As I traverse the files, I would load each applicable files for a feature into a `SourceFile` and call [manifest_items](https://github.com/web-platform-tests/wpt/blob/d44d043ebcc0453d4351244389ba30b2ea45e322/tools/manifest/sourcefile.py#L849C16-L849C16)

Let me know what you think about that

Reply:
Ah, that makes perfect sense, so simple! I was worried we'd have to find the right `MANIFEST.json` from some code outside of WPT, which would lead to some complications.